numpy/core/src/multiarray/mapping.c

```diff
@@ -703,7 +716,7 @@ prepare_index(PyArrayObject *self, PyObject *index,
     *ndim = new_ndim + fancy_ndim;
     *out_fancy_ndim = fancy_ndim;
 
-    if (make_tuple) {
+    if (index != index_raw) {
```

Comment:
And we eliminate `make_tuple` by simply checking if the fixed index is different from the original one.

Again, a small performance hit for a 32-bit comparison, but negligeable